phoenix-core-client/src/main/java/org/apache/phoenix/query/ConnectionQueryServices.java

```diff
@@ -99,22 +99,27 @@ public interface ConnectionQueryServices extends QueryServices, MetaDataMutated
     public TableDescriptor getTableDescriptor(byte[] tableName) throws SQLException;
 
     public HRegionLocation getTableRegionLocation(byte[] tableName, byte[] row) throws SQLException;
-    public List<HRegionLocation> getAllTableRegions(byte[] tableName) throws SQLException;
+
+    public List<HRegionLocation> getAllTableRegions(byte[] tableName, int queryTimeout)
```

Comment:
I would recommend not changing the public API but add a new API. We had similar problems in the past when we introduced a timeout in ScanningResultIterator. The existing APIs can send values like LONG.max.

Reply:
The clients using the API either have StatementContext object or they have Config object so either way they need to provide timeout anyway.
If we keep the API as is, anyone would only use that directly even though they might have PhoenixStatement level overridden timeout. So using public API without timeout is not good practice anyway right?

For ScanningResultIterator, the constructor has timeout param and the clients are either providing Long.MAX_VALUE or Context level overridden timeout but the pattern is same i.e. clients using the APIs are providing timeout.

Reply:
Sorry, it is TableResultIterator which broke when we modified its constructor. This was fixed via https://issues.apache.org/jira/browse/PHOENIX-7102

Reply:
The point is we need to maintain API compatibility

Reply:
I see, sounds good. If we maintain the old API contract, we still want it to figure out the timeout value from the config right?

Let me also mark the old API deprecated.

Reply:
That is even better. The old API can fetch the timeout from config instead of using Long.max. That would have saved you some test changes where you are doing exactly that.

Reply:
I think many of those test changes will be required because many of them don't have real config as such, we are either mocking or using constant. 

No problem, this can be taken care of in general. Though we have this test recently written (this month) that is getting stuck and not moving forward. This seems bit of a concern, either for using static test method directly on BaseResultIterator is becoming problematic or EnvironmentEdgeManager is causing troubles. Either way, the test might be blocking the whole build.